crates/cli/src/opts/build/core.rs

```diff
@@ -144,6 +145,10 @@ pub struct CoreBuildArgs {
     #[command(flatten)]
     #[serde(flatten)]
     pub project_paths: ProjectPathsArgs,
+
+    /// Profile to use for compiling the project.
+    #[arg(long)]
+    pub profile: Option<Profile>,
```

Comment:
I think we can have this in the global args and make it initialize `Config::selected_profile`